src/query_plan.c

```diff
@@ -80,19 +80,35 @@ static size_t serializeResult(QueryPlan *qex, SearchResult *r, RSSearchFlags fla
   return count;
 }
 
+/**
+ * Returns true if the query has timed out and the user has requested
+ * that we do not drain partial results.
+ */
+static int hasNoDrainTimeout(const QueryPlan *qex) {
```

Comment:
convert to a macro, this is in a very tight loop.

Reply:
Compilers will probably just inline it, but yes a macro is more appropriate.